Tests failing on NVIDIA Tesla T4, AWS G4 instance

Tests fail when using AWS G4 instance.

Steps to reproduce:
1. Set up CUDA 11.3 (latest) and NVIDIA Docker on a fresh EC2 instance of type g4dn.8xlarge.
2. Build the trition_fil Docker image: `docker build -t triton_fil -f ops/Dockerfile .`
3. Run the CI script: `LOCAL=1 ./qa/run_tests.sh`

When I switched the instance to p3.2xlarge type (V100 GPU), the tests run successfully. 

Error messages:
* `lightgbm` model
```
AssertionError: 
Arrays are not almost equal to 7 decimals
Mismatched elements: 1 / 1 (100%)
Max absolute difference: 0.36781818
Max relative difference: 0.36781818
 x: array([0.6321818], dtype=float32)
 y: array([1.], dtype=float32)
```
* `xgboost` model
```
AssertionError:                                                                                                                                                           
Arrays are not almost equal to 7 decimals                                                                                                                                 
Mismatched elements: 1 / 1 (100%)         
Max absolute difference: 2.                                                          
Max relative difference: 1.               
 x: array([0.], dtype=float32)                                                                                                                                            
 y: array([2.], dtype=float32)
```

--- a/src/triton_fil/triton_tensor.cuh
+++ b/src/triton_fil/triton_tensor.cuh
@@ -184,6 +184,12 @@ class TritonTensor {
   {
     auto head = reinterpret_cast<typename std::conditional<
         std::is_const<T>::value, const std::byte*, std::byte*>::type>(buffer);
+    auto cuda_res = cudaStreamSynchronize(stream_);
+    if (cuda_res != cudaSuccess) {
+      throw TritonException(
+          TRITONSERVER_errorcode_enum::TRITONSERVER_ERROR_INTERNAL,
+          "Failed stream synchronization in TritonTensor sync");
+    }
     for (auto& out_buffer : final_buffers) {
       try {
         raft::copy(

